server/models/book.js

```diff
@@ -0,0 +1,50 @@
+import { getObjectId, dummyData } from '../helpers/modelHelpers';
+
+export default class Book {
+    constructor(args) {
+      const fields = ['title', 'isbn', 'author', 'quantity', 'publishedYear'];
+      const errors = [];
+      const dataType = {title:'string', isbn:'number', author: 'string', quantity: 'number', publishedYear:'number'};
+      
+      // check if required fields are present
+      fields.forEach(field => {
+        if (!args[field]) {
+          errors.push({ path: field, message: `${field} is required` });
+        } else if (typeof(args[field]) !== dataType[field]) {
+          errors.push({ path: field, message: `${field} must be a ${dataType[field]}` });
+        } 
+      });
+      
+      for (let key in args) {
+        if (fields.indexOf(key) == -1) {
+          errors.push({ message: `invalid field: ${key} found in args` });
+        } else {
+          this[key] = args[key];        
+        }
+      }
+
+      if (errors.length > 0) {
+        // throw new Error (JSON.stringify(errors,null,4));
```

Comment:
remove comment

Reply:
Line 26 can be modified to be just if(errors.length)

Reply:
Noted